src/client/components/map/ProjectDistrictsMap.tsx

```diff
@@ -0,0 +1,110 @@
+/** @jsx jsx */
+import { Box, jsx, Spinner } from "theme-ui";
+import MapboxGL from "mapbox-gl";
+import React, { useEffect, useRef, useState } from "react";
+import bbox from "@turf/bbox";
+
+import { BBox2d } from "@turf/helpers/lib/geojson";
+
+import { ProjectNest } from "../../../shared/entities";
+import { DistrictGeoJSON } from "../../types";
+import { getDistrictColor } from "../../constants/colors";
+
+const style = {
+  mapContainer: {
+    height: "250px",
+    display: "inline-block",
+    position: "relative",
+    p: "15px"
+  },
+  map: {
+    position: "absolute",
+    top: 0,
+    bottom: 0,
+    left: 0,
+    right: 0
+  }
+} as const;
+
+const ProjectDistrictsMap = ({
+  project,
+  context
+}: {
+  readonly project: ProjectNest;
+  readonly context: "home" | "communityMaps";
+}) => {
+  const mapRef = useRef<HTMLDivElement>(null);
+  const [mapLoaded, setMapLoaded] = useState<boolean>(false);
+
+  // TODO #179 - the districts property can't be defined in the shared/entities.d.ts right now
+  // @ts-ignore
+  const districts: DistrictsGeoJSON | undefined = project.districts;
+
+  useEffect(() => {
+    if (mapRef.current === null) {
+      return;
+    }
+
+    districts.features.forEach((feature: DistrictGeoJSON, id: number) => {
+      // eslint-disable-next-line
+      feature.properties.color = getDistrictColor(id);
+    });
+
+    const bounds = districts && (bbox(districts) as BBox2d);
+    const map = new MapboxGL.Map({
+      container: mapRef.current,
+      style: {
+        version: 8,
+        sources: {},
+        layers: []
+      },
+      bounds,
+      fitBoundsOptions: { padding: 15 },
+      interactive: false
+    });
+
+    map.on("load", function() {
+      districts &&
+        map.addSource("districts", {
+          type: "geojson",
+          data: districts
+        });
+      map.addLayer({
+        id: "districts",
+        type: "fill",
+        source: "districts",
+        layout: {},
+        paint: {
+          "fill-color": { type: "identity", property: "color" }
+        }
+      });
+      map.resize();
+
+      setMapLoaded(true);
+    });
```

Comment:
React is showing a warning in the console re: this handler:
![image](https://user-images.githubusercontent.com/4432106/126794707-276ecfc2-27d1-4417-9436-3c05c66f08a9.png)

We need to turn off the `load` event handler if/when the React component gets unloaded - otherwise as the warning indicates we'll have a memory leak. In this case that includes the Mapbboxgl map, so this could potentially be a large performance concern

```suggestion
    function onLoad() {
      districts &&
        map.addSource("districts", {
          type: "geojson",
          data: districts
        });
      map.addLayer({
        id: "districts",
        type: "fill",
        source: "districts",
        layout: {},
        paint: {
          "fill-color": { type: "identity", property: "color" }
        }
      });
      map.resize();

      setMapLoaded(true);
    }
    map.on("load", onLoad);
    return () => {
      map.off("load", onLoad);
    };
```